src/Post/Post.js

```diff
@@ -441,19 +466,47 @@ export function Post({ datum, isExpanded, setIsExpanded, index, isFloating }) {
         'is-floating': isFloating,
       })}
     >
-      <div className={cx('post-content', { 'is-expanded': isExpanded })}>
+      <div
+        className={cx('post-content', {
+          'is-expanded': isExpanded,
+          'is-floating': isFloating,
+        })}
+      >
         <>
           <PostHeader
             city={'Baghdad'}
             user_name={datum?.user_name}
             user_thumbnail={datum?.user_thumbnail}
             datum={datum}
+            isExpanded={isExpanded}
           />
 
           <MediaSection
             media_items={datum?.media_items}
             isExpanded={isExpanded}
+            dimensions={dimensions}
+            mediaIndex={mediaIndex}
+            setMediaIndex={setMediaIndex}
           />
+          <section
+            className={cx('progress-dots-section', {
+              'is-expanded': isExpanded,
+            })}
+          >
+            {datum?.media_items.length > 1 && (
+              <div className="progress-dots">
+                {datum?.media_items.map((_, index) => (
+                  <div
+                    key={index}
+                    className={cx('progress-dot', {
+                      'is-active': index === mediaIndex,
+                      'is-expanded': isExpanded,
+                    })}
+                  ></div>
+                ))}
+              </div>
+            )}
+          </section>
```

Comment:
**100% optional**: I think this section should be moved inside `MediaSection`. So the state of `mediaIndex` can be inside `MediaSection` since only these two sections need it.

Reply:
They were already in mediaSeciton, we moved them because they were messing with our calculation.